phantazm-api/src/main/java/com/github/phantazmnetwork/api/RayUtils.java

```diff
@@ -4,29 +4,41 @@
 import net.minestom.server.coordinate.Point;
 import net.minestom.server.coordinate.Pos;
 import net.minestom.server.coordinate.Vec;
+import net.minestom.server.instance.Instance;
 import org.jetbrains.annotations.NotNull;
 
 import java.util.ArrayList;
+import java.util.Comparator;
 import java.util.List;
 import java.util.Optional;
 
+/**
+ * Utility methods related to ray tracing.
+ */
 public class RayUtils {
 
     private RayUtils() {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * Ray traces a {@link Shape} and finds an intersection position. This does not ray trace a {@link Shape}'s children.
```

Comment:
shouldn't this raytrace children? otherwise, stairs will have incorrect collision

Reply:
(nvm, just saw the lower method)